modules/multi2vec-palm/clients/palm.go

```diff
@@ -66,62 +66,110 @@ func (v *palm) VectorizeQuery(ctx context.Context, input []string,
 func (v *palm) vectorize(ctx context.Context,
 	texts, images, videos []string, config ent.VectorizationConfig,
 ) (*ent.VectorizationResult, error) {
+	var textEmbeddings [][]float32
+	var imageEmbeddings [][]float32
+	var videoEmbeddings [][]float32
 	endpointURL := v.getURL(config)
-	payload := v.getPayload(texts, images, videos, config)
+	maxCount := v.getMaxCount(texts, images, videos)
+	for i := 0; i < maxCount; i++ {
+		text := v.safelyGet(texts, i)
+		image := v.safelyGet(images, i)
+		video := v.safelyGet(videos, i)
+		payload := v.getPayload(text, image, video, config)
+		statusCode, res, err := v.sendRequest(ctx, endpointURL, payload)
+		if err != nil {
+			return nil, err
+		}
+		textVectors, imageVectors, videoVectors, err := v.getEmbeddingsFromResponse(statusCode, res)
+		if err != nil {
+			return nil, err
+		}
+		textEmbeddings = append(textEmbeddings, textVectors...)
+		imageEmbeddings = append(imageEmbeddings, imageVectors...)
+		videoEmbeddings = append(videoEmbeddings, videoVectors...)
+	}
+
+	return v.getResponse(textEmbeddings, imageEmbeddings, videoEmbeddings)
+}
+
+func (v *palm) safelyGet(input []string, i int) string {
+	if i < len(input) {
+		return input[i]
+	}
+	return ""
+}
+
+func (v *palm) getMaxCount(texts, images, videos []string) int {
+	count := len(texts)
+	if len(images) > count {
+		count = len(images)
+	}
+	if len(videos) > count {
+		count = len(videos)
+	}
+	return count
```

Comment:
I think this can be simplified to:
```go
func (v *palm) getMaxCount(texts, images, videos []string) int {
    return max(len(texts), len(images), len(videos))
}
```

Go introduced variadic `min`/`max` builtins with 1.21 🙂

Reply:
fixed! those are really nice methods!